### DISCUSSION TITLE
"Minimum Infill Length" behavior in Superslicer

### DISCUSSION THREAD

**Discussion by emailcraigs** [supermerill/SuperSlicer#3267] [Q&A] [answered]
I have used Simplify3d for awhile and have started using Superslicer 2.5.59 but have an issue. As as example, when printing a triangle, that comes up to a sharp point, SS will put infill into each layer as it gets smaller and smaller to the top point of the triangle. This is causing the top to overheat and deform. The slice in SS looks like this:
![SS](https://user-images.githubusercontent.com/31557493/192121106-52fa06f7-0c89-477c-9da1-df64d4b5a38e.jpg)

The only setting I have found that changes this is "solid infill threshold area". I set it to 0 and get this:
![SS_solid infill threshold at 0](https://user-images.githubusercontent.com/31557493/192121132-23d1afd5-7795-43ad-9c50-def6b40a116e.jpg)

But even this still deforms the small area from the infill. When using Simplify3d I am able to set "Minimum Infill Length" to 5 and get this, which does not melt the small area as it stops putting in infill at a certain size:
![S3D](https://user-images.githubusercontent.com/31557493/192121155-54f97cbb-8b3b-4182-aefa-d9e59de65588.jpg)

This shows the print results with the simplify3d print on the left and SS on the right:
![IMG_4135](https://user-images.githubusercontent.com/31557493/192121399-bc8c8c6d-37be-4863-a513-cb9b1e6014dd.jpg)

Is there a setting in SS that acts like the S3D setting "Minimum Infill Length" or is this a new feature I should request? This is important for printing small areas without having an overheating issue.

**Comment by neophyl** [comment]
Can you post your zipped up project file here so I can experiment with the settings (if the part isn’t proprietary that is). 
Usually deformed tops like that are caused by the filament slow down if layer time values. Setting those much lower can often stop this which is why I’d like the project to try.

**Reply by emailcraigs** [reply]
[Hollow_Pyramid_top test.zip](https://github.com/supermerill/SuperSlicer/files/9642632/Hollow_Pyramid_top.test.zip)

This is literally just the top of the hollow pyramid calibration part that I stripped down to be able to test the top part more quickly with various settings. The issue shows up perfectly there, but it would apply to  a lot of other small sections on various parts. I'm curious to see what you figure out. Thanks! I also have a simplify3d file if you can use it.

**Reply by neophyl** [reply]
Thanks for that.  I'll have a go this evening after I get back from work.

**Comment by neophyl** [comment, marked as answer]
Sorry for the delay, had a hectic week so far,
Heres a print with my default settings, tweaked ever so slightly to keep the small perimeters speed the same for the print (at 25mm/s).  I had to make sure the small perimeters speed on Print Settings was set suitably and also the Filament Settings>Cooling>Very Short layer time speeds matched.  This also had my cooling fan at 75% except for bridges and top layers so the cooling could have been increased but there was no need if you keep your speed constant.
![IMG_2253](https://user-images.githubusercontent.com/46955527/192828280-b52eed5c-9179-4471-b229-62edb172d6c8.JPG)

As you can see keeping the speed the same for all the perimeters works wonders. This is some end of roll cheap recycled PLA from 3djake.  

I'm sure I could get it slightly better if I played around and experimented further but from a cooling perspective I think its fine at the moment.

Heres my project settings if you want to compare.  There are probably a lot of differences due to the different printers.
[Hollow_Pyramid_top test_neo.zip](https://github.com/supermerill/SuperSlicer/files/9666928/Hollow_Pyramid_top.test_neo.zip)
